sockets/1.2/modules/ROOT/pages/sockets-documentation.adoc

```diff
@@ -594,7 +594,8 @@ Define your own custom protocol by writing a class that extends `TcpProtocol`.
 [[eof-protocol]]
 === Eof Protocol
 
-Reading is terminated by the stream being closed by the client.
+This protocol transfers data. The stream end is indicated by closing the client connection. Reading through this protocol is terminated by the client by shutting down its output stream, also known as TCP Half Close feature. Use this protocol with SSL sockets that use TLS 1.3 version and later, (Half Close feature was introduced in TLS 1.3).
```

Comment:
Is it possible to clarify these things? 
This protocol transfers data from xxx to yyy.
What closes the client connection? Is it the connector?
Is this right? The client reads through the protocol and shuts down its output stream when it is done. What output stream? How does shutting down an output stream stop the reading of data? Is it the client's output stream?
What is the Half Close feature - is it the client shutting down the output stream?

Reply:
When connecting using the EOF protocol, the client connects to a server and starts writing the message on the socket connection output stream. To signal the end of the message, the client is required to signal the end of file (EOF). This is accomplished closing the output stream and keeping the input stream open, awaiting for a response (This is known as half close). Then the server sends a response and closes it's OutputStream closing the whole socket connection.

For TLS secured connections, the use of TLS 1.3 and JDK 11 is required, since in TLS 1.2 and earlier versions, the half close is not supported (the half close triggers the close notify alert in order to prevent TLS Truncation attacks). TLS 1.3 is supported by JDK11 and newer versions.